weldx/geometry.py

```diff
@@ -1439,6 +1439,9 @@ def local_coordinate_system(
         relative_position = np.clip(relative_position, 0, 1)
 
         coordinates = np.array([1, 0, 0]) * relative_position * self._length
+        if isinstance(coordinates, pint.Quantity):
```

Comment:
So the basis vectors of the LCS have no unit? This confuses me,  I thought we are now demand units, and trigger a warning if not units are provided.

Reply:
This only applies to the LCS generated by a `LinearHorizontalTraceSegment` so more of an edge case that (I guess) is excluded for now because of tests/compatibility

Reply:
Due to the transition, the whole thing is currently a mess. As @CagtayFabry mentioned, this only applies to this segment class. The LCS currently supports both formats but the geometry classes are expect LCS without coordinates. The point of this PR was just to make LCS coordinate units possible without breaking anything. So as I mentioned earlier, there are some strange and ugly workarounds like this one. These will disappear in the next PR when I fix the whole package to be compatible with strict units.